src/hotspot/cpu/riscv/templateInterpreterGenerator_riscv.cpp

```diff
@@ -904,18 +904,19 @@ void TemplateInterpreterGenerator::lock_method(Register Rflags, Register Rscratc
              | return address  |   |                   |   | return address  |   |
              |   previous fp ------+                   +------ previous fp   |   |
              |       ...       |                           |       ...       |   |
-             |   argument 0    |                 locals--> |   argument 0    |   |
-             |       ...       |                           |       ...       |   |
-             |   argument n    |                           |   argument n    |   |
-     esp --> |       ___       |                           |-----------------|   |
+             +-----------------+                           +-----------------+   |
              |                 |                           |                 |   |
-             |                 |                           |     locals      |   |
-             |                 |                   fp -->  |                 |   |
-      sp --> |                 |                           +=================+   |
-             +=================+                           | return address  |   |
-               stack grows down                            |   previous fp ------+
-                      |                                    |-----------------|
-                      v                                    |                 |
+             |space for locals |                           |     locals      |   |
+             |                 |                           |                 |   |
+             +-----------------+                           |-----------------|   |
+             |   argument n    |                           |   argument n    |   |
+             |       ...       |                           |       ...       |   |
+      sp --> |   argument 0    |                   fp -->  |   argument 0    |   |
+     esp --> +=================+                           +=================+   |
+               stack grows down                            | return address  |   |
+                      |                                    |   previous fp ------+
+                      v                                    |-----------------|
+                                                           |                 |
```

Comment:
The issue is that you cannot reverse the argument order on expression stack -- the order is defined by java bytecode

```
anton@tornado:~$ cat Test.java 

public class Test {
        static int f1;
        static int f2;

        public static void f(int a, int b) {
                f1 = a;
                f2 = b;
        }

        public static void main(String args[]) throws Exception {
                f(1, 2);
        }
}

anton@tornado:~$ javap -c Test
Compiled from "Test.java"
public class Test {
  static int f1;

  static int f2;

  public Test();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method java/lang/Object."<init>":()V
       4: return

  public static void f(int, int);
    Code:
       0: iload_0
       1: putstatic     #2                  // Field f1:I
       4: iload_1
       5: putstatic     #3                  // Field f2:I
       8: return

  public static void main(java.lang.String[]) throws java.lang.Exception;
    Code:
       0: iconst_1
       1: iconst_2
       2: invokestatic  #4                  // Method f:(II)V
       5: return
}
```

Main method places first argument `1`, then second `2` in this exact order. So expression stack at the point of call is `..., 1, 2 ->` (in jvms notation).

Reply:
You're right